### ISSUE TITLE
Untranslatable strings

### ISSUE THREAD

**Issue by TotalCaesar659** [lpereira/hardinfo#38]
Hello! When I fixed Russian translation, I tried to add some untranslated strings to ru.po, converted to MO and launched the program with it. But the strings hasn't been translated, even if they was added in ru.po.
Here is the screenshots of some of that untranslatable places.
![2016-12-28-003048_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21510670/71b1ec8e-cca7-11e6-8d8a-22d9e999b79f.png)
![2016-12-28-003104_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21510673/76f0fd48-cca7-11e6-9e46-e64184b91c88.png)
![2016-12-28-003124_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21510675/7a6ab6b2-cca7-11e6-9cbd-1db1b1f1a3fd.png)
![2016-12-28-003144_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21510676/7d744170-cca7-11e6-903e-7efde17a2bff.png)

**Comment by lpereira** [comment]
Nice finds. I'll take a look later.

**Comment by mckaygerhard** [comment]
hi @TotalCaesar659 , please can u confirm that the memory amount are displayed correctly, in computer->Resume->Memory (please revise it in russian env, not in english) to confirm a bug in my translated po file or in code

**Comment by TotalCaesar659** [comment]
Hi @mckaygerhard, I don't see "Resume" in program. Which point I have to check on these pictures?
![2016-12-28-221951_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21529789/29cef1b8-cd4d-11e6-967c-7ce42bfae860.png)
![2016-12-28-222104_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21529794/31cbf49c-cd4d-11e6-86f4-805122f7b276.png)
I can reboot system to Russian and send a screenshot.

**Comment by mckaygerhard** [comment]
sorry my fault, in Computer->summary->memory , in spanish i cannot see the amount but in english amount of memory appears! can u confirm if in russnik appears correctly the translated stringa nd also the amount? also double-chek in devices->memory too, send screenshot if u can but its optionally

mi interes its that if the string memory are translated and shows amount of ram!

**Comment by TotalCaesar659** [comment]
Here is the screenshot in Russian.
Look at the Memory, it can't be translated, because the value disappears with it. There is the comment "//Load Average is not updated if locale is not C, switch locale to C" in /modules/computer.c (https://github.com/lpereira/hardinfo/blob/master/modules/computer.c). Maybe it has an influence to "Memory too".
![2016-12-28-233502_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21531252/8bf1ad04-cd57-11e6-8d74-41008383b7db.png)
![2016-12-28-233511_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21531256/8ff921ac-cd57-11e6-8e9b-60b2626f6a46.png)
![2016-12-28-233517_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21531259/95d43206-cd57-11e6-913c-b60139165407.png)

**Comment by mckaygerhard** [comment]
as i suspect, its not translatable, i have a patch to solve it (a little nasty but works) in the pull resquests 28 thanks for confirmation..

**Comment by lpereira** [comment]
Hello @TotalCaesar659 -- the last commit in HEAD should be slightly better in handling most of these fields. If it's still not working, please report back.

As for the memory fields, I'll take another look and report back.

**Comment by TotalCaesar659** [comment]
After commit https://github.com/lpereira/hardinfo/commit/907a6249077745aa6d19eb7280823dd769223a29, Hardinfo shows "Unknown field: Memory" in Russian.
![2016-12-30-194556_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21570987/febcf0a2-cedb-11e6-9fc7-2fcf5916fff6.png)
In English, it's alright.
![2016-12-30-195427_1280x974_scrot](https://cloud.githubusercontent.com/assets/14265316/21570977/e8350824-cedb-11e6-90ff-65f483db48ed.png)

**Comment by mckaygerhard** [comment]
how's the progress of this?

**Comment by mckaygerhard** [comment]
lasted commits from bp0 solved that issue, now can be closed! i tested in debian older squeeze : works, and debian newer jeesie/sid works!  

this issue can be closed!

the non translated its due are missing from pó file.. later i can translated

![hardinfomemetranslatable](https://user-images.githubusercontent.com/1571724/28135346-de9a055a-6713-11e7-81a9-fde4b63629bd.png)

**Comment by mckaygerhard** [comment]
now with #86 this issue are already fixed , close it!

**Comment by bp0** [comment]
All instances mentioned here are now translatable, but this work continues as part of #100.

**Comment by TotalCaesar659** [comment]
After translating latest master build, I found some bugs.
1) headbar is not full in translated version;
2) translated version has no counters of processors, their cores and threads;
![1](https://user-images.githubusercontent.com/14265316/29299204-af84bc98-8175-11e7-8ff2-d73044f2ecf6.png)
![2](https://user-images.githubusercontent.com/14265316/29299212-ba29f758-8175-11e7-9be7-a348d045c240.png)
3) there are many untranslatable strings left.
![2](https://user-images.githubusercontent.com/14265316/29299375-def15c4c-8176-11e7-9833-99b5175cd807.png)
![3](https://user-images.githubusercontent.com/14265316/29299459-674defba-8177-11e7-98d4-a59ac8736c40.png)
![4](https://user-images.githubusercontent.com/14265316/29299463-6bf33656-8177-11e7-9190-562c0fbe3bd8.png)
![5](https://user-images.githubusercontent.com/14265316/29299466-71162fda-8177-11e7-9565-38d22f2b17c2.png)

**Comment by mckaygerhard** [comment]
i have spanish and russian, i'll test russian and report feedback later..

**Comment by bp0** [comment]
@TotalCaesar659 

# Procs / Cores / Threads
Strings with the `c-format` tag need to have their `%s`, `%d`, `%.2f`, etc strings put in appropriate places.
So each string needs to have the `%d` where the number should go. I've done this now for this case, but please review it: https://github.com/lpereira/hardinfo/pull/151/commits/76e9013cb0f302e75a14aa47e0950d3711130bbe.
Also,
https://github.com/bp0/hardinfo/blob/76e9013cb0f302e75a14aa47e0950d3711130bbe/po/ru.po#L1847
is the string that pulls them together. So if something other than `;` is appropriate, please change it.

# Computer -> Summary
This is a messy one. The string you're looking for looks like this: `"<b>%s \342\206\222 Summary</b>"`

# Benchmark Results
The strings are stored in benchmark.conf translated! This is a known issue. See https://github.com/lpereira/hardinfo/pull/147#issuecomment-322165709

# Network stats
The strings are directly from netstat, and so there is not an easy way to translate them. They don't appear anywhere in the hardinfo code.

# Others
The OpenGL Renderer: `Unknown` string and the total memory `kB` string should be translatable, I'll look into that now.

**Comment by mckaygerhard** [comment]
i can confirm some of the inconsistences reported by @TotalCaesar659 but differs in spanish respect russian.. in russian i have same problems reported by @TotalCaesar659 

but in spanish i have strange behavior, in computer info the printing information sub-titile are duplicate:
![hardinfo-impresoras-duplicate](https://user-images.githubusercontent.com/1571724/29318101-af1eb642-819c-11e7-94f5-116539fe2a3e.png)

and:
![hardinfo-procesaor](https://user-images.githubusercontent.com/1571724/29318219-24c72028-819d-11e7-889c-b72188f4505c.png)

**Comment by mckaygerhard** [comment]
with lasted build today russian are fixed but spanish dont:
![hardinfo-russian](https://user-images.githubusercontent.com/1571724/29318379-bd2f9a70-819d-11e7-8993-026d564c9e2c.png)

**Comment by bp0** [comment]
> [spanish procs/cores/threads]

You haven't translated those strings yet:
https://github.com/lpereira/hardinfo/blob/master/po/es.po#L1803
They were added after your Spanish translation work was done.

> but in spanish i have strange behavior, in computer info the printing information [sub-title] are duplicate:

That is because of the recent change from printf()s to info_*() functions. It is not a translation related problem.

**Comment by mckaygerhard** [comment]
how i must do that:

#: modules/devices.c:151
#, c-format
msgid "%d physical processor"
msgid_plural "%d physical processors"
msgstr[0] ""
msgstr[1] ""

only put in the two followed lines and that's all?

**Comment by bp0** [comment]
msgstr[0] is the singular form, msgstr[1] is the plural form

**Comment by TotalCaesar659** [comment]
> Strings with the c-format tag need to have their %s, %d, %.2f, etc strings put in appropriate places.

Yeah, I've just forgot to add them. I have to sleep more.
And I've found more bugs:
1) Menu -> Help -> About modules. Descriptions are untranslatable, but they're in ![ru.po](https://github.com/lpereira/hardinfo/blob/master/po/ru.po).
![1](https://user-images.githubusercontent.com/14265316/29342116-a3ca7908-8231-11e7-8e4e-027159e3a6db.png)
![2](https://user-images.githubusercontent.com/14265316/29342119-a79ca786-8231-11e7-98c2-abeea41f250a.png)
![3](https://user-images.githubusercontent.com/14265316/29342121-a9fbfff4-8231-11e7-8104-01b2bd15d427.png)
![4](https://user-images.githubusercontent.com/14265316/29342122-ac0eb30e-8231-11e7-9d73-ec067048c371.png)

2) Look at windows' headbars.
These programs are called in English as "Computer module", "Network module" etc.
But some well-wisher created them as addition of two strings, "*"+"module". Of course, he hasn't thought about another languages and their rules of word combinations (e.g. cases).
So, it looks horrible in Russian, like "Компьютер модуль" instead of "Модуль компьютера", "Сеть модуль" instead of "Модуль сети" etc.
Hollywood-movie style as it is :-)

**Comment by bp0** [comment]
The descriptions are fixed by 9086f4f89b408e70476bb89a75e1b6de7b37faed.

The dialog title is composed with a format string "% Module". There are two ways we could fix it.
Easy way is to change it to something like "Module: %s" or "%s (Module)", which might make more sense in all languages. The harder way is to put the text "Module" in each module's name, and translate each one independently.

**Comment by TotalCaesar659** [comment]
It would be better to use the last way with separate names. Colon looks allogenic in the headbar.

**Comment by bp0** [comment]
I agree, but there will still be "About X", so it is still being constructed using English word order rules.

**Comment by TotalCaesar659** [comment]
@lpereira When I open my result in any benchmark, my CPU specs ("That Machine") are translated correctly.
![1](https://user-images.githubusercontent.com/14265316/74186311-6ba60a00-4c5b-11ea-9bb9-add26ab5feb3.png)
But other ones aren't.
![2](https://user-images.githubusercontent.com/14265316/74186336-7bbde980-4c5b-11ea-9442-e8f9f58a0642.png)
Looks like Hardinfo reads whole string from [benchmark.conf](https://github.com/lpereira/hardinfo/blob/master/data/benchmark.conf). Could you fix it?

**Comment by bp0** [comment]
Yes, the CPU config string is also stored, but actually gets retranslated, because the format is always the same, but the description is not. For example on arm, you might see it say 2x ARM Cortex-A72 + 4x ARM Cortex-A53, instead of packs/cores/threads.

**Comment by lpereira** [comment]
As part of converting benchmark.conf to JSON, I'll just store the numbers and generate the strings as we display them.  This will save some space in the server, too, and avoid translation issues (e.g. if the benchmark is submitted in, say, a French locale, it should display this information in Russian if the current locale is Russian.)

**Comment by bp0** [comment]
Yeah, alright, although the existing format actually does store them as
numbers already, but along with the CPU description string. I guess the
idea was that description is platform dependent. What does 6 cores, 6
threads tell you on that 2Big.4little machine? Do any ARM have SMT yet?
That kind of thing.

**Comment by TotalCaesar659** [comment]
@lpereira This strings consequence (346-350) can not be translated.
https://github.com/lpereira/hardinfo/blob/c9d5ec9fff4076e7c625f8d12ad011d3a6460482/modules/devices/storage.c#L346
